src/libcore/convert.rs

```diff
@@ -45,6 +69,12 @@ use marker::Sized;
 /// let s = "hello".to_string();
 /// is_hello(s);
 /// ```
+///
+/// # Generic Impls
+///
+/// - `AsRef` auto-dereference if the inner type is a reference or a mutable
+/// reference
```

Comment:
Oh for that I'd say that this is a transitive trait, e.g. if `T` can go to `U` and `U` can go to `V`, then `T` can go to `V` directly.